### ISSUE TITLE
XL: GetObject read error

### ISSUE THREAD

**Issue by krishnasrinivas** [minio/minio#1921]
curl of 200G object is erroring out.

```
krishna@minio2:~$ curl 'http://127.0.0.1:10000/testbucket/200gfile?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=7EK16BG19GRBIG1USWP2%2F20160617%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20160617T123257Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=0308ede6a8865f86e3512c85162c20f655d350f452d942528fd3c4f0543737e9' | md5sum
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0  200G    0 25.0M    0     0  9364k      0  6:13:15  0:00:02  6:13:13 9363k
  4  200G    4 9200M    0     0  15.2M      0  3:43:17  0:10:01  3:33:16 13.0M
curl: (18) transfer closed with 205101465600 bytes remaining to read
322b345f96a9d316d5294d2d0ddbd4cb  -
krishna@minio2:~$ 
```

**Comment by harshavardhana** [comment]
Can you validate again with #1910 merged ?

**Comment by hackintoshrao** [comment]
@harshavardhana : Running the tests.

**Comment by harshavardhana** [comment]
Fixed by #1959
